File: Week_7-ReactJs/30_march_2026/Assessment_03/shopping-cart/src/components/CartItem.js
import React from 'react';

function CartItem({ item, onUpdateQuantity, onRemoveItem }) {
  return (
    <div className="cart-item">
      <div className="cart-item-info">
        <span className="cart-item-name">{item.name}</span>
        <span className="cart-item-subtotal">${item.price * item.quantity}</span>
      </div>

      <div className="cart-item-controls">
        <button onClick={() => onUpdateQuantity(item.id, item.quantity - 1)}>−</button>
        <span className="quantity">{item.quantity}</span>
        <button onClick={() => onUpdateQuantity(item.id, item.quantity + 1)}>+</button>

        {/* Remove single item */}
        <button className="remove-btn" onClick={() => onRemoveItem(item.id)} title="Remove item">
          🗑️ Remove
        </button>
      </div>
    </div>
  );
}

export default CartItem;

File: Week_12/9_May_2026/Assessment/FoodOrderingSystem/FoodOrderingSystem/src/FoodOrderingSystem.Core/Entities/CartItem.cs
namespace FoodOrderingSystem.Core.Entities;

public class CartItem
{
    public int Id { get; set; }
    public int CartId { get; set; }
    public int FoodItemId { get; set; }
    public int Quantity { get; set; }

    public Cart Cart { get; set; } = null!;
    public FoodItem FoodItem { get; set; } = null!;

    public decimal SubTotal => FoodItem != null ? FoodItem.Price * Quantity : 0;
}
